lib/v2/xiaohongshu/util.js

```diff
@@ -65,7 +66,104 @@ const getBoard = (url, cache) =>
         false
     );
 
+const getNotes = (url, cache) =>
+    cache.tryGet(
+        url + '/notes', // To avoid mixing with the cache for `user.js`
+        async () => {
+            let user = '';
+            let notes = [];
+
+            const browser = await require('@/utils/puppeteer')({ stealth: true });
+            try {
+                const page = await browser.newPage();
+                logger.debug(`Requesting ${url}`);
+                await page.goto(url);
+
+                const [otherInfo, userPosted] = await Promise.all(
+                    ['/api/sns/web/v1/user/otherinfo', '/api/sns/web/v1/user_posted'].map((p) =>
+                        page
+                            .waitForResponse((res) => {
+                                const req = res.request();
+                                return req.url().includes(p) && req.method() === 'GET';
+                            })
+                            .then((r) => r.json())
+                    )
+                );
+
+                // Get full text for each note
+                const notesPromise = userPosted.data.notes.map(async (n) => {
+                    const noteUrl = url + '/' + n.note_id;
+                    const notePage = await browser.newPage();
+                    logger.debug(`Requesting ${noteUrl}`);
+                    await notePage.goto(noteUrl);
+
+                    const content = await notePage.content();
+                    const feed = JSON.parse(
+                        content
+                            .match(/<script>window.__INITIAL_STATE__=({.*})<\/script>/g)[0]
+                            .match(/{.*}/g)[0]
+                            .replace(/undefined/g, '""')
+                    );
```

Comment:
You can obtain the JavaScript object directly by running
```js
const feed = await notePage.evaluate(() => window.__INITIAL_SSR_STATE__);
```
instead of parsing the text.